model/textparse/interface.go

```diff
@@ -78,28 +80,65 @@ type Parser interface {
 	Next() (Entry, error)
 }
 
-// New returns a new parser of the byte slice.
-//
-// This function always returns a valid parser, but might additionally
-// return an error if the content type cannot be parsed.
-func New(b []byte, contentType string, parseClassicHistograms, skipOMCTSeries bool, st *labels.SymbolTable) (Parser, error) {
+// extractMediaType returns the mediaType of a required parser. It tries first to
+// extract a valid and supported mediaType from contentType. If that fails,
+// the provided fallbackType (possibly an empty string) is returned, together with
+// an error. fallbackType is used as-is without further validation.
+func extractMediaType(contentType, fallbackType string) (string, error) {
 	if contentType == "" {
-		return NewPromParser(b, st), nil
+		if fallbackType == "" {
+			return "", errors.New("Non-compliant scraper sending blank Content-Type and no fallback_scrape_protocol specified for target")
+		}
+		return fallbackType, errors.New("Non-compliant scraper sending blank Content-Type, using fallback_scrape_protocol for target")
 	}
 
+	// We have a contentType, parse it.
 	mediaType, _, err := mime.ParseMediaType(contentType)
 	if err != nil {
-		return NewPromParser(b, st), err
+		if fallbackType == "" {
+			retErr := errors.New("Cannot parse Content-Type and no fallback_scrape_protocol for target")
+			return "", errors.Join(retErr, err)
+		}
+		retErr := errors.New("Could not parse received Content-Type, using fallback_scrape_protocol for target")
+		return fallbackType, errors.Join(retErr, err)
+	}
+
+	// We have a valid media type, either we recognise it and can use it
+	// or we have to error.
+	switch mediaType {
+	case "application/openmetrics-text", "application/vnd.google.protobuf", "text/plain":
+		return mediaType, nil
+	}
+	// We're here because we have no recognised mediaType.
+	if fallbackType == "" {
+		return "", fmt.Errorf("Scraper sending unrecognisable Content-Type '%q' and no fallback_scrape_protocol specified for target", contentType)
 	}
+	return fallbackType, fmt.Errorf("Content-Type '%q' not recognised mediaType, using fallback_scrape_protocol for target", contentType)
```

Comment:
See above. Also, we are conflating mediaType and ContentType now. Maybe let's use exactly the same error message as above, just that now we should include the fallback picked.
Maybe like this:
```suggestion
	return fallbackType, fmt.Errorf("received unsupported Content-Type %q, using fallback_scrape_protocol %q", contentType, fallbackType)
```

Reply:
Done!